SDC/index.js

```diff
@@ -1,46 +1,96 @@
-const { Client } = require( 'pg' );
-const express = require( 'express' );
-let app = express();
-
+require('newrelic');
+const express = require('express');
+const bodyParser = require('body-parser');
+const path = require('path');
+const dotenv = require('dotenv');
+const Review = require('./Review.js');
+const pool = require( './pgsql.js' );
+// const { Pool } = require( 'pg' );
+// const cass = require( './cassie.js' );
+// const Sequelize = require('sequelize');
 
-let connectionString = 'postgres://postgres:postgres@localhost:5432/testdb2020';
+/******************* */
+// const pool = new Pool({
+//   user: 'postgres',
+//   host: 'localhost',
+//   database: 'testdb2020',
+//   password: 'postgres',
+//   port: 5432,
+// })
+// console.log(pool.password)
+// pool.connect()
+// .catch( ( err ) => {
+//   console.log( err )
+// } )
+/******************** */
 
-const client = new Client({
-  connectionString: connectionString
-})
+dotenv.config();
+// const sequelize = new Sequelize('user_reviews', `${process.env.DB_USERNAME}`, `${process.env.DB_PASSWORD}`, { dialect: 'mysql', dialectOptions: { multipleStatements: true } });
 
-client.connect();
+const app = express();
+const port = process.env.PORT || 3000;
 
-app.set( 'port', process.env.PORT || 4000 );
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use('/', express.static(path.join(__dirname, 'client', 'dist')));
+app.use('/products', express.static(path.join(__dirname, 'client', 'dist')));
+app.use((req, res, next) => {
+  res.header('Access-Control-Allow-Origin', '*');
+  next();
+});
 
-app.get( '/', ( req, res, next ) => {
-  client.query('SELECT * FROM Employee WHERE id = $1', [1], ( err, result ) => {
-    if ( err ) {
-      console.log( err );
-      res.status( 400 ).send( err );
-    }
-    res.status( 200 ).send( result.rows );
-  });
+app.get('/products/*', (req, res) => {
+  res.sendFile(path.join(__dirname, 'client', 'dist', 'index.html'));
 });
 
-app.listen( 4000, () => {
-  console.log( 'Server is running.. on Port 4000' );
+app.get('/api/testing/test', ( req, res ) => {
+  pool.getUsers( req, res );
 });
 
+/*http://localhost:3000/api/reviews/products/:productId*/
+/*SEND WITH...........*/
+app.get('/api/reviews/products/:productId', ( req, res ) => {
+   pool.getUsers( req, res, req.params.productId );
+})
 
-// const Sequelize = require('sequelize');
-// const mysql = require('mysql2/promise');
-// const dotenv = require('dotenv');
+app.get('/api/reviews/products/all',
+  ( req, res ) => se)
+
+app.post('/api/reviews/products/:productId',
+  ( req, res ) => {
+    let insertable = {
+      product_id: req.body.product_id, // INT
+      user: req.body.user, // STRING
+      score: req.body.score, // INT
+      title: req.body.title, // STRING
+      body: req.body.body, // STRING
+      recommend: true, // BOOL
+      date: '10/15/2020',
```

Comment:
Is there another way to handle the data here? A hardcoded date is probably not best practice.

Reply:
Same with 'recommend', unless true is your default value